### PR TITLE
connect_no_timeout: Add comments for the reason of the loop

### PR DESCRIPTION
**Pull request by keithnoguchi**
We want to purge all the invalid connections, e.g. due to the
network failure and recreate all the connections fresh in a
single connection request, so that we can get the connection
in the next attempt.  And also, once all those invalid
connections were purged, the next connect_no_timeout just block
to create a new connection, as there is no valid connections
in the pool.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -172,6 +172,13 @@ impl<C: ManageConnection + Send> Pool<C> {
             return self.try_get_connection().await;
         }
 
+        // We go through the max size loop here in case there are no valid
+        // connections in the pool, e.g. due to the network disconnect, so
+        // that we can immediately purge stale connections and initiate the
+        // new connections in one shot.
+        //
+        // Note that since we just returns error, in case no valid connections,
```
Comment (nate-onesignal) [inline src/lib.rs:180]: I'd call out here that we return an error if we're unable to create a *new* connection after the first one is invalid

Comment (keithnoguchi) [reply src/lib.rs]: Thanks @nate-onesignal and I've updated the comment, as I understand the logic better now than before.  Please review.

File: src/lib.rs [hunk 2]
```diff
@@ -172,6 +172,19 @@ impl<C: ManageConnection + Send> Pool<C> {
             return self.try_get_connection().await;
         }
 
+        // We go through the max size loop here to:
+        //
+        // 1) Get a new connection.
+        // 2) Grab a connection from the pool.
+        //
+        // In case 1, it will wait in the future and either get the
+        // new connection or return error.  In case 2, we check the validify
```
Comment (niels-s) [inline src/lib.rs:181]:
```suggestion
        // new connection or return error.  In case 2, we check the validity
```

Comment (keithnoguchi) [reply src/lib.rs]: Thank you, Niels, for catching this up.  Please review the updated version.

**Pull request merged**